### ISSUE TITLE
Not possible to verify CSR signature

### ISSUE THREAD

**Issue by tlhackque** [gknocke/Crypt-PKCS10#11]
The API doesn't expose the binary asn.1 of the SIGNED section of the CSR.  Further, for ECC, the signature is an ECDSA object that needs to be parsed into its components for verification.

Both of these are addressed in my sandbox.  certificationRequest() returns the SIGNED section of the CSR, and signature(2) will return the components of an ECDSA signature.

Tests provide examples of verifying CSRs signed with RSA and DSA keys.

Verifying ECDSA signatures should be possible with the API provided, but I haven't found a Perl module that has reasonable inputs.  So the test for that is incomplete.

Verifying CSRs is not the responsibility of Crypt::PCKS10, and an API to do the verification will not be provided.  The caller is responsible for using the signature algorithm to determine what procedure to use, and the other methods to get the requisite data.

**Comment by tlhackque** [comment]
Test for signatures on EC CSRs now implemented.

**Comment by tlhackque** [comment]
Fixed in V1.7

Signatures are verified by default by new().  This can be suppressed and the signature verified manually (or ignored.)

RSA, DSA and ECC keys are supported, with MD2, MD4, MD5 and SHA hashes.

examples/verify_csr will provide a go/no-go verification using Crypt::PKCS10.
examples/dump_csr will provide more detail.

I went further than planned because the mechanics of the general case are a bit involved, and it makes sense to do it once.
